src/components/AuthContainer.js

```diff
@@ -26,8 +26,7 @@ const AuthContainer = ({ children }) => {
     [seenAzurePreview === false && isAzureUser(), () => h(AzurePreview)],
     [registrationStatus === undefined && !isPublic, authspinner],
     [registrationStatus === userStatus.unregistered, () => h(Register)],
-    [userNeedsToAcceptTos === undefined && !isPublic, authspinner],
-    [userNeedsToAcceptTos && name !== 'privacy', () => h(TermsOfService)],
+    [displayTosPage && _.isUndefined(termsOfService.userContinuedUnderGracePeriod) && name !== 'privacy', () => h(TermsOfService)],
```

Comment:
Should this be `displayTosPage || _.isUndefined(termsOfService.userContinuedUnderGracePeriod)`?

Reply:
the idea here is that userContinuedUnderGracePeriod is only set when the user clicks the continue button -- so we dont want to force display the TOS page if they havent been to it yet. It will just show the alert in the top right at that point.

Reply:
Makes sense.